src/app/services/api.ts

```diff
@@ -58,8 +58,15 @@ export class ApiService {
   }
 
   public async downloadDocument(document: Document): Promise<void> {
-    console.log(document);
-    const blob = await this.downloadResource(document._id);
+    let blob;
+    try {
+      blob = await this.downloadResource(document._id)
+    } catch (e) {
+      throw new Error(e)
+    }
+    if (!blob) {
+      return
```

Comment:
Do you want to throw an error here too? The download would be failing at this point.

Reply:
good point, yeah I will throw an Error here instead of return

Reply:
Added in a throw on null blob